tiledb/sm/serialization/query.cc

```diff
@@ -677,13 +677,7 @@ Status index_read_state_from_capnp(
     fragment_indexes.emplace_back(tile_idx, cell_idx);
   }
 
-  auto read_state = reader->read_state();
-  if (read_state == nullptr) {
-    throw std::runtime_error(
-        "index_read_state_from_capnp: read_state within SparseIndexReaderBase "
-        "object is null.");
-  }
-  *read_state = tiledb::sm::SparseIndexReaderBase::ReadState(
+  reader->read_state() = tiledb::sm::SparseIndexReaderBase::ReadState(
```

Comment:
Unstatus this function and make it return a ReadState. The top level should use a setter to set the read state. Document the setter and mention it's only and should be only used for deserialization.